agent/http_test.go

```diff
@@ -417,62 +418,59 @@ func TestHTTPAPI_TranslateAddrHeader(t *testing.T) {
 func TestHTTPAPIResponseHeaders(t *testing.T) {
 	t.Parallel()
 	a := NewTestAgent(t, `
+		ui_config {
+			# Explicitly disable UI so we can ensure the index replacement gets headers too.
+			enabled = false
+		}
 		http_config {
 			response_headers = {
 				"Access-Control-Allow-Origin" = "*"
 				"X-XSS-Protection" = "1; mode=block"
+				"X-Frame-Options" = "SAMEORIGIN"
 			}
 		}
 	`)
 	defer a.Shutdown()
 
+	requireHasHeadersSet(t, a, "/v1/agent/self")
+
+	// Check the Index page that just renders a simple message with UI disabled
+	// also gets the right headers.
+	requireHasHeadersSet(t, a, "/")
+}
+
+func requireHasHeadersSet(t *testing.T, a *TestAgent, path string) {
+	t.Helper()
+
 	resp := httptest.NewRecorder()
-	handler := func(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-		return nil, nil
-	}
+	req, _ := http.NewRequest("GET", path, nil)
+	a.srv.handler(true).ServeHTTP(resp, req)
 
-	req, _ := http.NewRequest("GET", "/v1/agent/self", nil)
-	a.srv.wrap(handler, []string{"GET"})(resp, req)
+	hdrs := resp.Header()
+	require.Equal(t, "*", hdrs.Get("Access-Control-Allow-Origin"),
+		"Access-Control-Allow-Origin header value incorrect")
 
-	origin := resp.Header().Get("Access-Control-Allow-Origin")
-	if origin != "*" {
-		t.Fatalf("bad Access-Control-Allow-Origin: expected %q, got %q", "*", origin)
-	}
+	require.Equal(t, "1; mode=block", hdrs.Get("X-XSS-Protection"),
+		"X-XSS-Protection header value incorrect")
 
-	xss := resp.Header().Get("X-XSS-Protection")
-	if xss != "1; mode=block" {
-		t.Fatalf("bad X-XSS-Protection header: expected %q, got %q", "1; mode=block", xss)
-	}
+	require.Equal(t, "*", hdrs.Get("Access-Control-Allow-Origin"),
```

Comment:
Why is Access-Control-Allow-Origin checked twice?

Reply:
🍝